app/controllers/assignments_controller.rb

```diff
@@ -43,13 +43,11 @@ class AssignmentsController < ApplicationController
 
   # GET /courses/1/assignments
   def index
-    
     # check whether is it a guest user 
-    @user = User.find_by_id(session[:user_id]) 
-    @guest_user = GuestUsersDetail.find_by_user_id(@user.id)
+    @guest_details = @course.find_guest_user_details(session[:user_id])
     # obtain assignment to be shown if is a guest user
-    if (@guest_user)
-      @assignment_to_be_shown = @guest_user.assignment_id
+    if @guest_details.length > 0
```

Comment:
What is guest_details.length? Number of assignments that the guest user participates in?  Can you document this somewhere?